### DISCUSSION TITLE
spacy svg file vs. dependency input

### DISCUSSION THREAD

**Discussion by zarakolagar120** [explosion/spaCy#8701] [Help: Other Questions] [answered]
Hi,
I have a question regarding the dependency input for the visualizer, I see that when one wants to render the data manually, one has to provide 'start' and 'end'  tokens for each arc. 
I was trying to see if I could parse the .svg output to go back to the dep input, however, this information is missing from the .svg file. So, my question is whether there is a way to get this information regarding the 'start' and 'end' tokens through parsing the svg output, or any other approach?
thank you very much in advance

**Comment by ines** [comment, marked as answer]
Hi! There's not really an easy way to convert the exported `.svg` back to the original token format – in order to generate the arcs, the heads/deps information is converted to cubic bezier curves, and the tokens are converted to `<text>` elements. You can see the implementation here: 

https://github.com/explosion/spaCy/blob/f9fd2889b79495d2c4607ad3c89438aacb5d24ec/spacy/displacy/render.py#L36

_In theory_, you could of course write a function that reverses the exact calculation here, but it's probably a hassle. But if you don't have the original input, you could still reconstruct it from the visualization: the words are the tokens, and their start and end character offsets are the character positions are the offset + the length of the word + whitespace. The head of a token is the index of the token  that points to it via the arrow. And the dependency label is the label of that arrow.

**Reply by zarakolagar120** [reply]
thank you very much for the detailed explanation :)
